masim.c

```diff
@@ -14,6 +14,16 @@
 #include "misc.h"
 #include "masim.h"
 
+// hugepage definitions
+#define PROTECTION (PROT_READ | PROT_WRITE)
+
+/* Only ia64 requires this */
+#ifdef __ia64__
```

Comment:
This macro is self-explaining it's only for ia64.  I'd prefer to remove the above comment.